### ISSUE TITLE
`depth_to_normals` returns all-zero "unit normals" when `H = 1` or `W = 1`, and `nan` in float16

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4398]
`depth_to_normals` documents a unit normal per pixel. When either spatial axis has length 1 the result is a zero vector in float32 and float64, and `nan` in float16 — silently, with the documented output shape.

## Repro

```python
import torch
from kornia.geometry.depth import depth_to_normals

K = torch.tensor([[[100.0, 0.0, 1.0], [0.0, 100.0, 0.0], [0.0, 0.0, 1.0]]])
for shape in [(1, 1, 1, 3), (1, 1, 3, 1), (1, 1, 1, 1), (1, 1, 2, 3)]:
    for dt in (torch.float32, torch.float16, torch.float64):
        n = depth_to_normals(torch.full(shape, 2.0, dtype=dt), K.to(dt))
        print(shape, str(dt)[6:], tuple(n.shape), n.norm(dim=1).flatten().float().tolist())
```

Executed on `main` `147ecc4e`, torch 2.14.0 (cpu; mps gives the same values):

| depth shape | float32 | float16 | float64 |
|---|---|---|---|
| `(1, 1, 1, 3)` (H = 1) | `(1, 3, 1, 3)`, norms `[0, 0, 0]` | `nan` everywhere | norms `[0, 0, 0]` |
| `(1, 1, 3, 1)` (W = 1) | `(1, 3, 3, 1)`, norms `[0, 0, 0]` | `nan` everywhere | norms `[0, 0, 0]` |
| `(1, 1, 1, 1)` | `(1, 3, 1, 1)`, norm `[0]` | `nan` | `[0]` |
| `(1, 1, 2, 3)` (control) | norms all `1.0` | `1.0` | `1.0` |

A flat plane at `H = 3` returns `[0, 0, 1]` at the centre, so the sign convention itself is fine; only the degenerate sizes are wrong.

## Mechanism

`spatial_gradient` along a length-1 axis is identically zero, so one of the two tangent vectors is zero, `torch.linalg.cross` returns zero, and `F.normalize(normals, dim=-1, p=2)` divides `0` by `max(‖0‖, 1e-12)`: `0` in float32/float64, and in float16 the `1e-12` floor underflows to `0`, giving `0 / 0 = nan`. Nothing checks `H >= 2` / `W >= 2`; `KORNIA_CHECK_SHAPE(depth, ["B", "1", "H", "W"])` accepts any positive size.

## Classification

**Broken output** — a zero "unit normal" (or `nan`) with the documented shape and no error. Idiom 4 (a degenerate input silently produces garbage instead of raising) and idiom 2 (the `1e-12` normalisation floor is not scaled to the working dtype).

## Disposition

n/a — single symbol. The focused fix is to reject `H < 2 or W < 2` with a `ShapeError` (a normal field needs two tangent directions); if a zero normal is the intended answer for the degenerate case, say so in the docstring and use a dtype-scaled floor so float16 returns zeros too. Either way the `test_convention_*` pin goes on the chosen contract.

Found in the batch-5 audit (row `5c-nm-05`, tracking issue #4262), recorded there as unfiled; the `W = 1` case originally reached this function through #4278's squeeze and now reaches it on its own after #4298.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
